File: LeetCode-Java/Remove Nth Node From End of List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        ListNode node1 = null, node2 = head, node3 = head;
        while(n > 0)
        {
            node3 = node3.next;
            n --;
        }
        while(node3 != null)
        {
            node1 = node2;
            node2 = node2.next;
            node3 = node3.next;
        }
        if(node1 == null)
        {
            return head.next;
        }
        else
        {
            node1.next = node2.next;
            return head;
        }
        
        
    }
}

File: Leetcode-cpp/remove_nth_node_from_end_of_list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        ListNode* fake_head = new ListNode(-1);
        fake_head->next = head;
        head = fake_head;
        ListNode* fast = fake_head;
        n++;
        while (n > 0) {
            fast = fast->next;
            n--;
        }
        while (fast != nullptr) {
            head = head->next;
            fast = fast->next;
        }
        head->next = head->next->next;
        return fake_head->next;
    }
};